Sources/StreamVideo/Controllers/CallController.swift

```diff
@@ -679,50 +682,127 @@ class CallController: @unchecked Sendable {
+
+        let sessionID = webRTCClient.sessionID
+        executeOnMain { [weak self] in
+            self?.call?.state.sessionId = sessionID
+            self?.call?.update(recordingState: response.call.recording ? .recording : .noRecording)
+            self?.call?.state.ownCapabilities = response.ownCapabilities
+            self?.call?.state.update(from: response)
+        }
+
+        setupStatsTimer()
+
+        return response
+    }
+
+    private func _makeWebRTCClient() -> WebRTCClient {
+        let webRTCClient = WebRTCClient(
+            user: user,
+            apiKey: apiKey,
+            //                response.credentials.server.url,
+            //                response.credentials.server.wsEndpoint,
+            //                response.credentials.token,
+            callCid: callCid(from: callId, callType: callType),
+            sessionID: UUID().uuidString,
+            //                response.ownCapabilities,
+            videoConfig: videoConfig,
+            //                response.call.settings.audio,
+            environment: .init(),
+            authenticator: .init(self)
+        )
+
+//        webRTCClient
+//            .connectionPublisher
+//            .sink { [weak self] in self?.handleSignalChannelConnectionStateChange($0) }
+//            .store(in: disposableBag)
+
+//        webRTCClient.onSignalConnectionStateChange = { [weak self] state in
+//            self?.handleSignalChannelConnectionStateChange(state)
+//        }
+//        webRTCClient.onSessionMigrationEvent = { [weak self] in
+//            self?.handleSessionMigrationEvent()
+//        }
+//        webRTCClient.onSessionMigrationCompleted = { [weak self] in
+//            self?.call?.update(reconnectionStatus: .connected)
+//        }
+
+        return webRTCClient
+    }
 }
 
-extension CallController {
-    struct Environment {
-        var webRTCBuilder: (
-            _ user: User,
-            _ apiKey: String,
-            _ hostname: String,
-            _ webSocketURLString: String,
-            _ token: String,
-            _ callCid: String,
-            _ sessionID: String?,
-            _ ownCapabilities: [OwnCapability],
-            _ videoConfig: VideoConfig,
-            _ audioSettings: AudioSettings,
-            _ environment: WebSocketClient.Environment
-        ) -> WebRTCClient = {
-            WebRTCClient(
-                user: $0,
-                apiKey: $1,
-                hostname: $2,
-                webSocketURLString: $3,
-                token: $4,
-                callCid: $5,
-                sessionID: $6,
-                ownCapabilities: $7,
-                videoConfig: $8,
-                audioSettings: $9,
-                environment: $10
-            )
+extension CallController: WebRTCClientDelegate {
+    func webRTClientMigrating(_ webRTCClient: WebRTCClient) {
+        Task { @MainActor [weak self] in
+            self?.call?.state.reconnectionStatus = .migrating
```

Comment:
why is this one different than the rest?

Reply:
Just copied what we had (and different handling was there already). Unified it for now